python/tests/scripts/d4_selfmod.metta

```diff
@@ -43,7 +46,7 @@
    (nop (change-state greeted True)))
 
 ;!(answer (listen (Hello)))
-!(answer (listen (My name is "Alexey")))
+!(answer (listen (My name is "Alexey"))) ;doesn't work correctly now due to caching?
```

Comment:
Yes, probably, I made some simple check for grounded sub-expression, but it is not enough. Anything grounded in calculation trace should prevent caching.